### DISCUSSION TITLE
Filename template with date TAKEN (not created)

### DISCUSSION THREAD

**Discussion by jetmcquack** [RhetTbull/osxphotos#1310] [Q&A] [answered]
Hello, i was looking for a software like osxphotos for years, so thanks in advance!
I'm looking to export all my photos with a filename structure like

YYYY-MM-DD - Title (or filename).ext

reading the guide i've found that there's a {created.date} substitution which may works for my date, but i can't understand if it's going to write the photo FILE creation date or the correct date of the photo i have fixed in Photos. I have a library of many '800s and beginning of '900s which i need to export like:

1895-06-30 - Marconi sending first radio signal.heic

so... is {created.date} the correct substitution for my case?

**Comment by RhetTbull** [comment]
Yes, the created date is the correct date. It is whatever date is shown in Photos so if you've adjusted the date, osxphotos will use the adjusted date.

**Comment by RhetTbull** [comment, marked as answer]
> YYYY-MM-DD - Title (or filename).ext

This would work for `--filename` to get the format you want:

 "{created.date} - {title?{title},{original_name}}"

This template uses the photo creation date (again, not the file creation date, but the date associated with the photo) and then checks to see if the title is set `title?` and if so, uses the title, and if not, uses the original filename.

You can test templates using the `osxphotos template` command which gives you an interactive template editor with tab completion that will render the template for the currently selected photo in Photos.

**Reply by jetmcquack** [reply]
thanks so much!!!!

**Reply by msg43** [reply]
THANKS!
